src/software/ai/navigator/path_planner/hrvo/simulated_hrvo_test.py

```diff
@@ -0,0 +1,462 @@
+import pytest
+from software.simulated_tests.simulated_test_fixture import (
+    simulated_test_runner,
+    pytest_main,
+)
+from software.simulated_tests.robot_enters_region_and_stops import *
+from software.simulated_tests.avoid_collisions import *
+import software.python_bindings as tbots
+from proto.ssl_gc_common_pb2 import Team
+from proto.message_translation.tbots_protobuf import create_world_state_from_state
+
+
+def get_zig_zag_params(front_wall_x, robot_y_delta, num_walls):
+    """
+    Gets the test params to cause movement in a zig zag pattern
+    due to multiple walls of enemy robots
+
+    :param front_wall_x: the x position of the first wall
+    :param robot_y_delta: the vertical separation between each robot in the wall
+    :param num_walls: the number of walls to use
+    :return: positions and destinations of friendly and enemy robots to cause
+             a zig zag movement
+    """
+    return (
+        [Point(x_meters=front_wall_x - 0.5, y_meters=0)],
+        [Point(x_meters=front_wall_x + 3 + 1.5, y_meters=0)],
+        [],
+        [
+            Point(x_meters=x_meters, y_meters=float(y_meters) / 10)
+            for x_meters in range(front_wall_x, front_wall_x + num_walls + 1, 1)
+            for y_meters in range(
+                0,
+                (1 if x_meters % 2 == 0 else -1) * 3 * int(robot_y_delta * 10),
+                (1 if x_meters % 2 == 0 else -1) * int(robot_y_delta * 10),
+            )
+        ],
+        [],
+        12,
+    )
+
+
+def get_robot_circle_pos(radius, num_robots, start):
+    """
+    Gets the test params to position robots in a circle and have them move
+    along each diameter
+
+    :param radius: the radius of the circle
+    :param num_robots: the number of robots in the circle
+    :param start: if True, start positioning the robots from (radius, 0)
+                  else, start from (-radius, 0)
+                  basically outputs robot start and end points based on boolean
```

Comment:
Curious why this is necessary?

Reply:
so this basically means that calling the function with start == True gives you the start positions of the robots. And calling it with start == False gives you the end positions of the same robots, which are directly opposite each start position.

this is basically so that the same function can be used for both. the boolean rotates the circle of positions generated by 180 degrees